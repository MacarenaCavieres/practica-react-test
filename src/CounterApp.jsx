import { PropTypes } from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const handleClick = () => {
        setCounter(counter + 1);
    };

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={handleClick}>+1</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};
