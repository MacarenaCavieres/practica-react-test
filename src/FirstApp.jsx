// const newMessage = {
//     message: "Hola mundo",
//     title: "New World",
// };

// const getResult = (a, b) => {
//     return a + b;
// };

import PropTypes from "prop-types";

export const FirstApp = ({ title = "no hay titulo", subtitle = "No hay subtitulo" }) => {
    return (
        <>
            <h2>Hola mundo</h2>
            {/* <code>{JSON.stringify(newMessage)}</code> */}
            {/* <h3>{getResult(4, 8)}</h3> */}
            <h1 data-testid="test-title">{title}</h1>
            <h4>{subtitle}</h4>
            <h4>{subtitle}</h4>
            <p>Primer componente</p>
        </>
    );
};

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    // subtitle: PropTypes.string.isRequired,
};

// FirstApp.defaultProps = {
//     title: "No hay titulo",
//     subtitle: "de nuevo no hay subtitulo",
// };
