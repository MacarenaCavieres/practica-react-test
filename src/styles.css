html,
body {
    background-color: #21232a;
    color: #ffffff;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    padding: 15px;
    border-radius: 10px;
    border: none;
    width: 100px;
    cursor: pointer;
}
