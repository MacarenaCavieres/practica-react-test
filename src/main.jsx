import React from "react";
import ReactDOM from "react-dom/client";
// import App from "./App";
import { FirstApp } from "./FirstApp";
// import { CounterApp } from "./CounterApp";
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <App /> */}
        {/* <FirstApp title="Titulo principal" /> */}
        <CounterApp value={5} />
    </React.StrictMode>
);
