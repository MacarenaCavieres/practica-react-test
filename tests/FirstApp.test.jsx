import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <FirstApp />", () => {
    test("Debe de hacer match con el snapshot", () => {
        const title = "Titulo titulo";
        const subtitle = "subt";
        const { container } = render(<FirstApp title={title} subtitle={subtitle} />);

        expect(container).toMatchSnapshot();
    });

    test("debe mostrar el titulo en un h1", () => {
        const title = "Nuevo titulo principal";
        const { container, getByText, getByTestId } = render(<FirstApp title={title} />);

        // expect(getByText(title)).toBeTruthy();

        // const h1 = container.querySelector("h1");
        // expect(h1.innerHTML).toContain(title);

        expect(getByTestId("test-title").innerHTML).toBe(title);
    });

    test("debe mostrar el subtitulo enviado por props", () => {
        const title = "Tremendo titulo";
        const subtitle = "Nuevo subtitulo";
        const { getByText } = render(<FirstApp title={title} subtitle={subtitle} />);

        expect(getByText(subtitle)).toBeTruthy();
    });
});
