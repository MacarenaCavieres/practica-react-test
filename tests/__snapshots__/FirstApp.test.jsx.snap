// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola mundo
  </h2>
  <h1
    data-testid="test-title"
  >
    Titulo titulo
  </h1>
  <h4>
    subt
  </h4>
  <h4>
    subt
  </h4>
  <p>
    Primer componente
  </p>
</div>
`;
